[Integration]: Fireworks ai Integration

bala-ceg opened the pull request:
## Description
Created Fireworks.ai ML Integration.

Fixes #9043 

## Type of change

- [x] ⚡ New feature (non-breaking change which adds functionality)


## Verification Process

To ensure the changes are working as expected:

 - [x]  Test Location: Please refer Testing Screenshots section
 - [x]   Verification Steps: Created the integration engine, verified all the data queries successfully mentioned in README.md
 

## Additional Media:

- [x] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.


## Testing Screenshots:
<img width="1648" alt="Screenshot 2024-04-24 at 11 16 01 PM" src="https://github.com/mindsdb/mindsdb/assets/70808619/3a4bfbb0-96e7-4fe1-9c5f-d8a771cfa22e">
<img width="1728" alt="Screenshot 2024-04-24 at 11 17 05 PM" src="https://github.com/mindsdb/mindsdb/assets/70808619/950dd6e2-d7f6-425c-acc1-b56472aae022">
<img width="1728" alt="Screenshot 2024-04-24 at 11 17 55 PM" src="https://github.com/mindsdb/mindsdb/assets/70808619/793df596-6abe-41ed-a06f-6079b134e0a1">
<img width="1728" alt="Screenshot 2024-04-24 at 11 18 34 PM" src="https://github.com/mindsdb/mindsdb/assets/70808619/ab2812d3-4eb3-407d-bc49-67eca7e64677">

bala-ceg commented:
@quest-bot loot #9043

ZoranPandovski commented on mindsdb/integrations/handlers/fireworks_handler/config.py:
```diff
@@ -0,0 +1,8 @@
+supported_task = {
```
Maybe change this to something more meaningfull too as: conversational or image mode?

bala-ceg commented on mindsdb/integrations/handlers/fireworks_handler/README.md:
```diff
@@ -0,0 +1,153 @@
+
+Before proceeding, ensure the following prerequisites are met:
+
+1. Install MindsDB [locally via Docker](https://docs.mindsdb.com/setup/self-hosted/docker) or use [MindsDB Cloud](https://cloud.mindsdb.com/).
```
FIxed it

bala-ceg commented on mindsdb/integrations/handlers/fireworks_handler/config.py:
```diff
@@ -0,0 +1,8 @@
+supported_task = {
```
Fixed it

bala-ceg commented on mindsdb/integrations/handlers/fireworks_handler/fireworks_handler.py:
```diff
@@ -0,0 +1,150 @@
+        args = self.model_storage.json_get("args")
+
+
+        response = self.connection.ChatCompletion.create(
```
Fixed it

bala-ceg commented:
@ZoranPandovski I have added embeddings and implemented all the review changes except adding tests.
Regarding tests, will it be suitable if just validate the query is being executed correctly rather validating the output

parthiv11 commented:
@bala-ceg are you working on this?

bala-ceg commented:
@parthiv11 you can work on this handler, if you want.

parthiv11 commented:
@ZoranPandovski only unit test left, right?

and should I proceed with new PR with this code ?

ZoranPandovski commented:
Let's continue with the work to the existing branch @parthiv11 and start with adding the unit tests

parthiv11 commented:
@ZoranPandovski as I dont have access to [bala-ceg:bala_fireworks_handler](https://github.com/bala-ceg/mindsdb/tree/bala_fireworks_handler), it will not be possible to make changes in this PR

bala-ceg commented:
@ZoranPandovski can i continue my work on this PR, if you are okay with it ? 
c.c: @parthiv11

parthiv11 commented:
Yes, if you want to continue. I don't have any issues

On Wed, Oct 2, 2024, 5:20 PM Balaji Seetharaman ***@***.***>
wrote:

> @ZoranPandovski <https://github.com/ZoranPandovski> can i continue my
> work on this PR, if you are okay with it ?
> c.c: @parthiv11 <https://github.com/parthiv11>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mindsdb/mindsdb/pull/9131#issuecomment-2388450282>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASBGDTFC3AKM7XN4O3GZQV3ZZPMZZAVCNFSM6AAAAABIS4IRRKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGOBYGQ2TAMRYGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

bala-ceg commented:
@ZoranPandovski the code is ready for your review
